### ISSUE TITLE
player reported that opp made a play and it didn't show up on their screen, and they timed out

### ISSUE THREAD

**Issue by domino14** [domino14/liwords#277]
```
ElJefe Found an error- computer was counting down an opponent or so I thought- coas
ElJefe The game ended and it gave me a loss!
ElJefe for timeout
ElJefe It didn’t register that he had made a play on my screen
```

**Comment by domino14** [comment]
If they disconnected from the socket, upon reconnection it should give them the full game history again. Is it possible that this is the "at most once" delivery of NATS? This is bound to happen occasionally. I had a situation myself a long time ago where I entered a couple of chats and they didn't show up. We might need to make game events at-least-once (with acks)....

**Comment by domino14** [comment]
```
2020-11-20T21:43:31.452-05:00 (api) - ElJefe disconnected (left-site)
2020-11-20T21:44:17.503-05:00 (api) - ElJefe came back
2020-11-20T21:50:37.983-05:00 (api) - ElJefe disconnected 
2020-11-20T21:50:38.180-05:00 (api) - rp302 plays ZA
2020-11-20T21:55:09.517-05:00 (api) - rp302 sends a message via chat
2020-11-20T21:55:50.591-05:00 (api) - game ends, ElJefe loses on timeout
2020-11-20T21:56:28.483-05:00 (api) - rp302 leaves
2020-11-20T22:03:11.244-05:00 (api) - ElJefe rejoins and gets GameHistory with message that he lost
```

**Comment by domino14** [comment]
client-side ping should fix this (detects connection is gone and reconnects / refreshes)

**Comment by domino14** [comment]
Sometimes players type in messages into chat and get no data back :(

In `liwords-socket`:

```go
_, message, err := c.conn.ReadMessage()
```

Logs show
```
{
    "level": "debug",
    "error": "read tcp 172.17.0.3:8087->10.0.1.216:2312: i/o timeout",
    "time": "2020-11-22T21:27:16Z",
    "message": "other-error-breaking-out"
}
```

So this disconnects them. I'm not sure whether the front-end detects this and tries reconnecting.

**Comment by domino14** [comment]
hopefully fixed by #288
